drivers/snmp-ups.c

```diff
@@ -1156,7 +1156,20 @@ static struct snmp_pdu **nut_snmp_walk(const char *OID, int max_iteration)
 			snmp_free_pdu(response);
 			break;
 		} else {
-			numerr = 0;
+			/* Checked the "type" field of the returned varbind if it is a type error
+			 * exception (only applicable with SNMPv2 or SNMPv3 protocol) */
+			if (response->variables->type == SNMP_NOSUCHOBJECT ||
+			    response->variables->type == SNMP_NOSUCHINSTANCE ||
+			    response->variables->type == SNMP_ENDOFMIBVIEW) {
+				upslogx(LOG_WARNING, "[%s] Warning: type error exception (OID = %s)",
+						upsname?upsname:device_name, OID);
+				snmp_free_pdu(response);
+				break;
```

Comment:
Should it set some numerr here?

Reply:
For me, it is not really an error, it just that the OID is not implemented. For large request, it should find the next available OID in the list without stopping loop treatment.

Reply:
FYI, this numerr is a mess.
it's supposed to catch transient fetch error, while in fact this is generally a missing OID.
I once had a patch to fix that, that got lost...